src/main/java/com/google/api/codegen/advising/Adviser.java

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.advising;
+
+import com.google.api.codegen.ConfigProto;
+import com.google.api.tools.framework.model.Diag;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.SimpleLocation;
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.List;
+
+public class Adviser {
+  private static final ImmutableList<AdviserRule> DEFAULT_RULES =
+      ImmutableList.<AdviserRule>of(new LanguageSettingsRule());
+
+  private final List<AdviserRule> rules;
+
+  private final Collection<String> suppressors;
+
+  public Adviser(Collection<String> suppressors) {
+    this(DEFAULT_RULES, suppressors);
+  }
+
+  public Adviser(List<AdviserRule> rules, Collection<String> suppressors) {
+    this.rules = rules;
+    this.suppressors = suppressors;
+  }
+
+  public void advise(Model model, ConfigProto configProto) {
+    for (AdviserRule rule : rules) {
+      if (suppressors.contains(rule.getName())) {
+        continue;
+      }
+
+      for (String message : rule.collectAdvice(model, configProto)) {
+        model
+            .getDiagCollector()
+            .addDiag(Diag.warning(SimpleLocation.TOPLEVEL, "(%s) %s", rule.getName(), message));
```

Comment:
Can we introduce a new `class GapicConfigLocation implements Location` instead of using `TOPLEVEL`? It would be really cool if we could eventually provide filename + line number in the warnings (not necessarily in this PR)

Reply:
We can do that if we return a config source from the `MultiYamlReader` and change everywhere that uses it.